Throw better error for missing arguments on publish script

## 🐛 Bug Report

I am trying to publish my site on Windows and `yarn run publish-gh-pages` and `npm run publish-gh-pages` fail in the same place

I have push access to the repository and can push from the command line. 

### Have you read the [Contributing Guidelines on issues](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

## To Reproduce

1. Initialize a new repo on windows
2. Try to publish the repository 

## Expected behavior

The site would be published

## Actual Behavior

Received this error (similar error with yarn as well)

```
 create mode 100644 sitemap.xml
error: cannot spawn sh: No such file or directory
error: cannot spawn sh: No such file or directory
error: failed to execute prompt script (exit code 255)
fatal: could not read Password for 'https://alexdriedger@github.com': No error
Error: Git push failed
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @ publish-gh-pages: `docusaurus-publish`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @ publish-gh-pages script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

## Reproducible Demo

(Paste the link to an example repo, including a `siteConfig.js`, and exact instructions to reproduce the issue.)

https://github.com/alexdriedger/SlayTheSpireModding

```
dowload the repo (https://github.com/alexdriedger/SlayTheSpireModding) and unzip
create a new github repo
git init
git add -A
git push -u origin master
yarn run publish-gh-pages
```

--- a/v1/lib/publish-gh-pages.js
+++ b/v1/lib/publish-gh-pages.js
@@ -54,6 +54,13 @@ if (!PROJECT_NAME) {
   shell.exit(0);
 }
 
+if (USE_SSH !== 'true' && !GIT_USER) {
+  shell.echo(
+    "Missing git user. Did you forget to export the 'GIT_USER' environment variable?",
+  );
+  shell.exit(0);
+}
+
 let remoteBranch;
 if (USE_SSH === 'true') {
   remoteBranch = `git@${GITHUB_HOST}:${ORGANIZATION_NAME}/${PROJECT_NAME}.git`;

